tests/components/mobile_app/test_webhook.py

```diff
@@ -236,3 +264,52 @@ def store_event(event):
     assert state.attributes["longitude"] == 2.0
     assert state.attributes["gps_accuracy"] == 10
     assert state.attributes["altitude"] == -10
+
+
+async def test_webhook_enable_encryption(hass, webhook_client, create_registrations):
+    """Test that encryption can be added to a reg initially created without."""
+    webhook_id = create_registrations[1]["webhook_id"]
+
+    enable_enc_resp = await webhook_client.post(
+        "/api/webhook/{}".format(webhook_id), json={"type": "enable_encryption"},
+    )
+
+    assert enable_enc_resp.status == 200
+
+    enable_enc_json = await enable_enc_resp.json()
+    assert len(enable_enc_json) == 1
+    assert CONF_SECRET in enable_enc_json
+
+    key = enable_enc_json["secret"]
+
+    enc_required_resp = await webhook_client.post(
+        "/api/webhook/{}".format(webhook_id), json=RENDER_TEMPLATE,
+    )
+
+    assert enc_required_resp.status == 400
+
+    enc_required_json = await enc_required_resp.json()
+    assert "error" in enc_required_json
+    assert enc_required_json["success"] is False
+    assert enc_required_json["error"]["code"] == "encryption_required"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4989696)